source/DurationPicker.js

```diff
@@ -0,0 +1,516 @@
+				// headerContainer required to avoid bad scrollWidth returned in RTL for certain text widths (webkit bug)
+				{name: 'headerContainer', classes: 'moon-expandable-list-item-header moon-expandable-picker-header moon-expandable-datetime-header', components: [
+					{name: 'header', kind: 'moon.MarqueeText'}
+				]},
+				{name: 'currentValue', kind: 'moon.MarqueeText', classes: 'moon-expandable-picker-current-value'}
+			]},
+			{name: 'drawer', kind: 'enyo.Drawer', resizeContainer: false, classes: 'moon-expandable-list-item-client indented', components: [
+				{name: 'client', kind: 'enyo.Control', classes: 'enyo-tool-decorator moon-date-picker-client', onSpotlightLeft: 'closePicker', onSpotlightSelect: 'closePicker'}
+			]}
+		],
+
+		/**
+		* @private
+		*/
+		bindings: [
+			{from: '.disabled', to: '.$.headerWrapper.disabled'},
+			{from: '.disabledPicker', to: '.$.hour.disabled'},
+			{from: '.disabledPicker', to: '.$.minute.disabled'},
+			{from: '.disabledPicker', to: '.$.second.disabled'}
+		],
+
+		/**
+		* @private
+		*/
+		constructor: function() {
+			this.inherited(arguments);
+			this.valueArray = {
+				'hour': 0,
+				'minute': 0,
+				'second': 0
+			};
+		},
+
+		/**
+		* @private
+		*/
+		create: function () {
+			this.inherited(arguments);
+			this.initDefaults();
+		},
+
+		/**
+		* @private
+		*/
+		initDefaults: function () {
+			this.setupPickers();
+			this.templateChanged();
+			if(this.value !== '') {
+				this.valueChanged();
+				this.countdownChanged();
+			} else {
+				this.noneTextChanged();
+			}
+		},
+
+		/**
+		* creating the hour, minute and second picker components
+		*
+		* @private
+		*/
+		setupPickers: function () {
+			this.createComponent(
+				{name: 'hourPicker', classes: 'moon-date-picker-wrap', components:[
+					{kind: 'moon.IntegerPicker', name: 'hour', min: 0, max: 23, wrap: true, digits: 2, value: this.valueArray['hour']},
+					{name: 'hourLabel', content: this.hourText, classes: 'moon-date-picker-label moon-divider-text'}
+				]}
+			);
+			this.createComponent(
+				{name: 'minutePicker', classes: 'moon-date-picker-wrap', components:[
+					{kind: 'moon.IntegerPicker', name: 'minute', classes: 'moon-date-picker-field', min: 0, max: 59, wrap: true, digits: 2, value: this.valueArray['minute']},
+					{name: 'minuteLabel', content: this.minuteText, classes: 'moon-date-picker-label moon-divider-text'}
+				]}
+			);
+			this.createComponent(
+				{name: 'secondPicker', classes: 'moon-date-picker-wrap', components:[
+					{kind: 'moon.IntegerPicker', name: 'second', classes: 'moon-date-picker-field', min: 0, max: 59, wrap: true, digits: 2, value: this.valueArray['second']},
+					{name: 'secondLabel', content: this.secondText, classes: 'moon-date-picker-label moon-divider-text'}
+				]}
+			);
+		},
+
+		/**
+		* @fires moon.DurationePicker#onChange
+		* @private
+		*/
+		handleChangeEvent: function (sender, ev) {
```

Comment:
`(inSender, inEvent)` is the old naming convention we no longer use. `(sender, ev)` is the current, shorter names.